src/sync/mpsc/tip.rs

```diff
@@ -0,0 +1,135 @@
+use std::sync::{Arc, Mutex};
+
+use task::{self, Task};
+use {Sink, Stream, AsyncSink, Async, Poll, StartSend};
+
+/// Tip is very similar to unbounded channel but only stores last value sent
+///
+/// I.e. if you want to send some value between from producer to a consumer
+/// and if consumer is slow it should may skip old values, the tip is
+/// a structure for the task.
+
+/// The transmission end of a channel which is used to send values
+///
+/// If the receiver is not fast enough only the last value is preserved and
+/// other ones are discarded.
+#[derive(Debug)]
+pub struct TipSender<T> {
+    inner: Arc<Mutex<Inner<T>>>,
```

Comment:
One trick I've done in the past is for all senders to store a `Weak` and the receiver to store an `Arc`. That way there's no need for an extra `closed` flag and all the races and such should in theory be taken care of.